### DISCUSSION TITLE
Lab6 合成error (Failed run(s) : 'design_1_caravel_0_0_synth_1')

### DISCUSSION THREAD

**Discussion by gene5487** [bol-edu/HLS-SOC-Discussions#165] [Q&A] [answered]
大家好，我在執行source run_vivado時，遇到了以下error，不知道應該如何解決? 謝謝。

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/58682521/cf999f57-b379-497a-a025-0f9e93379f89)

**Comment by alexlu0414** [comment, marked as answer]
Hi同學，可以試試看在[vvd_caravel_fpga_40mhz.tcl]這個檔案大約200行的位置加上bram.v試試看，有可能是合成時檔案有少。
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/121178881/cf1eccc6-3248-4d6c-816f-2f0bf00fe898)

**Comment by samlo123** [comment]
可以去.run資料夾看錯誤檔案的狀況。
如果檔案都有include，且都沒寫錯，但還是崩潰，有可能是虛擬機的記憶體不夠。
可以重啟虛擬機，記憶體使用最少的情況下試試看。

**Reply by gene5487** [reply]
查看lab-wlos_baseline/vivado/vvd_caravel_fpga/vvd_caravel_fpga.runs/design_1_caravel_0_0_synth_1/runme.log後，發現是tcl裡面的set files沒有包含到bram，加上後已成功合成。謝謝！
